pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -397,9 +397,12 @@ if [[ -z $_NIXOS_REBUILD_REEXEC && -n $canRun && -z $fast ]]; then
             SHOULD_REEXEC=1
         fi
     else
-        runCmd nix "${flakeFlags[@]}" build --out-link "${tmpDir}/nixos-rebuild" "$flake#$flakeAttr.config.system.build.nixos-rebuild" "${extraBuildFlags[@]}" "${lockFlags[@]}"
-        if p=$(readlink -e "${tmpDir}/nixos-rebuild"); then
-            SHOULD_REEXEC=1
+        targetSystem=$(runCmd nix eval --raw "$flake#$flakeAttr.pkgs.system")
+        if [[ $(nix show-config --json | jq -r '.system.value') == $targetSystem ]]; then
```

Comment:
It seems that all other `nix` invocations are made via `runCmd`, so we should do it here too.
```suggestion
        if [[ $(runCmd nix show-config --json | jq -r '.system.value') == $targetSystem ]]; then
```